packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -29,9 +41,23 @@ export type QueuedRequestControllerEvents =
 export type QueuedRequestControllerActions =
   QueuedRequestControllerEnqueueRequestAction;
 
+type AllowedActions =
+  | NetworkControllerGetStateAction
+  | NetworkControllerSetActiveNetworkAction
+  | NetworkControllerSetProviderTypeAction
+  | NetworkControllerGetNetworkClientByIdAction
+  | NetworkControllerFindNetworkClientIdByChainIdAction
+  | SelectedNetworkControllerSetNetworkClientIdForDomainAction
+  | AddApprovalRequest;
+
+export type QueuedRequestMessenger = ControllerMessenger<
```

Comment:
This type seems to be used in the middleware file. What do you think about moving it to that file instead of here? It's also slightly odd that this is not a RestrictedControllerMessenger, but I may be confused about when we should be using a restricted messenger vs. not.

Reply:
Renamed as [`QueuedRequestMiddlewareMessenger`](https://github.com/MetaMask/core/pull/1970/files#diff-02c32f437b122b5c71f0ac6bdf304cd22f6d4569c9c9b8ecbb711c11c98cd0d3R17-R20) type and moved to `QueuedRequestMiddleware` file!

The reason this is defined using `ControllerMessenger` rather than `RestrictedControllerMessenger` is that it's modeling messengers defined by other controllers that are in the middleware chain.

Reply:
I have more to say on this, but since this is outdated, I will resolve this conversation and start a new one.